packages/smart-contracts/test/contracts/BatchConversionErc20Payments.test.ts

```diff
@@ -0,0 +1,586 @@
+      feeAddress: string,
+    ): Promise<ContractTransaction> => {
+      return useBatchRouter
+        ? testBatchConversionProxy.connect(_signer).batchRouter(
+            [
+              {
+                paymentNetworkId: '0',
+                conversionDetails: convDetails,
+                cryptoDetails: {
+                  tokenAddresses: [],
+                  recipients: [],
+                  amounts: [],
+                  paymentReferences: [],
+                  feeAmounts: [],
+                },
+              },
+            ],
+            feeAddress,
+          )
+        : testBatchConversionProxy
+            .connect(_signer)
+            .batchERC20ConversionPaymentsMultiTokens(convDetails, feeAddress);
+    };
+  };
+
+  /**
+   * Function used to check the events emitted from the batch conversion proxy.
+   * @dev referenceExample and feeAddress are not args because there values never change
+   */
+  const emitOneTx = (
+    result: Chai.Assertion,
+    _convDetail: ConversionDetail,
+    _conversionToPay = conversionToPay,
+    _conversionFees = conversionFees,
+    _testErc20ConversionProxy = testErc20ConversionProxy,
+  ) => {
+    return result.to
+      .emit(_testErc20ConversionProxy, 'TransferWithConversionAndReference')
+      .withArgs(
+        _convDetail.requestAmount,
+        ethers.utils.getAddress(_convDetail.path[0]),
+        ethers.utils.keccak256(referenceExample),
+        _convDetail.feeAmount,
+        '0',
+      )
+      .to.emit(_testErc20ConversionProxy, 'TransferWithReferenceAndFee')
+      .withArgs(
+        ethers.utils.getAddress(DAI_address),
+        ethers.utils.getAddress(_convDetail.recipient),
+        _conversionToPay,
+        ethers.utils.keccak256(referenceExample),
+        _conversionFees,
+        feeAddress,
+      );
+  };
+
+  /**
+   * @notice update convDetail, do an ERC20 conversion batch payment with a single payment inside and calculate the balances
+   * @param path to update the convDetail
+   */
+  const onePaymentBatchConv = async (path: string[]) => {
+    await getConvToPayAndConvDetail(to, path, amountInFiat, feesAmountInFiat, 0, chainlinkPath);
+
+    const result = batchConvFunction([convDetail], feeAddress);
+    await emitOneTx(expect(result), convDetail, conversionToPay, conversionFees);
+    if (logGas) {
+      const tx = await result;
+      await tx.wait(1);
+      const receipt = await tx.wait();
+      console.log(`gas consumption: `, receipt.gasUsed.toString());
+    }
+
+    [fromDiffBalanceExpected, toDiffBalanceExpected, feeDiffBalanceExpected] =
+      expectedERC20Balances([conversionToPay], [conversionFees], batchConvFee);
+  };
+
+  /**
+   * @notice generate nTimes 2 convDetails, do an ERC20 conv batch payment with theses 2*nTimes payments
+   *         and calculate the balances
+   * @param path2 to update the second convDetail
+   */
+  const manyPaymentsBatchConv = async (path2: string[], nTimes: number) => {
```

Comment:
This method is only called with `nTimes = 1`, and it's quite hard to understand it. It's worth a simplification.

Maybe you can simplify the global structure of these tests by removing the dependency on the aggregator. Define a few fixed conversion rates (e.g 1.1, 2.2 and 3.3), set the oracles and you don't need a method to compute expected amounts anymore.